src/main/java/frc/robot/Limelight.java

```diff
@@ -22,4 +61,168 @@ public void togglePipeline() {
+        
+        /*
+         * Gets the x & y values from the robotPositionValues array
+         */
+        double robotPositionX = getRobotPosition3D()[0].doubleValue();
+        double robotPositionY = getRobotPosition3D()[1].doubleValue();
+
+        /*
+         * Uses the x & y values to construct a translation2d, then returns it
+         */
+        Translation2d limelightTranslation2d = new Translation2d(robotPositionX, robotPositionY);
+        return limelightTranslation2d;
+
+    }
+
+    public Transform2d createTransform2D(Translation2d constructorTranslation2d, Rotation2d constructorRotation2d) {
+        /*
+         * Uses a translation2d & a rotation2d parameter to construct a transform2d, then returns it
+         */
+        Transform2d limelightTransform2d = new Transform2d(constructorTranslation2d, constructorRotation2d);
+        return limelightTransform2d;
+
+    }
+
+    /*
+     * returns a boolean value that lets us know if the limelight has any targets
+     */
+    public boolean hasTargets() {
+        return table.getEntry("tv").getBoolean(false);
+    }
+
+    /*
+     * Uses the vertical distance from and the angle from the limelight to the low target to calculate the horizontal distance
+     * Will need to be updated later as the height and angle are currently unknown
+     */
+    public double getHorizontalFromLow()
+    {
+        double verticalOffset = getVerticalOffset();
+
+        if(verticalOffset > 0 && verticalOffset <= Constants.LimelightConstants.MAX_VERT_OFFSET_FOR_LOW)
+        {
+            double horizontalFromLow = Constants.LimelightConstants.HEIGHT_TO_LOW / Math.tan(verticalOffset);
+            return horizontalFromLow;
+        }
+        
+        return 0.0;
+    }
+
+    /*
+     * Uses the vertical distance from and the angle from the limelight to the mid target to calculate the horizontal distance
+     * Will need to be updated later as the height and angle are currently unknown
+     */
+    public double getHorizontalFromMed()
+    {
+        double verticalOffset = getVerticalOffset();
+
+        if(verticalOffset > MAX_VERT_OFFSET_FOR_LOW && verticalOffset <= MAX_VERT_OFFSET_FOR_MED)
+        {
+            double horizontalFromMed = HEIGHT_TO_MED / Math.tan(verticalOffset);
+            return horizontalFromMed;
+        }
+
+        return 0.0;
+    }
+
+    /*
+     * Uses the vertical distance from and the angle from the limelight to the high target to calculate the horizontal distance
+     * Will need to be updated later as the height and angle are currently unknown
+    */
+    public double getHorizontalFromHigh()
+    {
+        double verticalOffset = getVerticalOffset();
+
+        if(verticalOffset > MAX_VERT_OFFSET_FOR_MED && verticalOffset <= MAX_VERT_OFFSET_FOR_HIGH)
+        {
+            double horizontalFromHigh = HEIGHT_TO_HIGH / Math.tan(verticalOffset);
+            return horizontalFromHigh; 
+        }
+
+        return 0.0;
+    }
```

Comment:
can there be one method so that other subsystems don't have to do limelight logic to figure out current target